streaming-payments.md

```diff
@@ -0,0 +1,375 @@
+is a bit simpler; when a request comes in it sends headers, and
+then the body will be sent letter-by-letter, as payments come in. The `Pay` header contains 3 parts:
+
+* amount (in this case, the price of one letter, in XRP)
+* user-specific destination address
+* a Base64-encoded conditionSeed
+
+Note how we are appending `'.' + user` to the shop's Interledger address! This is a special feature of Interledger
+addresses, they can be subnetted endlessly, just add another `.` at the end to convert an account address
+to a ledger prefix, and then add new sub-accounts after that. In this case, we want to know which user is paying
+for letters, so by telling each user a different Interledger sub-address, we can neatly keep our users apart.
+
+When a transfer comes in, the server opens the [ILP packet](https://github.com/interledgerjs/ilp-packet).
+In there is a destination account and a destination amount, which will usually (if all went well) be equal
+to the amount of the transfer. In later tutorials, we will see how one ILP payment can be chained together
+in multiple transfers, but for now, we will only use single-transfer payments.
+
+Whereas in the previous tutorial, the fulfillment was generated randomly, here it's generated deterministically from
+the combination of the shop's interledger address, the user's userId, this payment's paymentId, and the
+conditionSeed (as a buffer) that was sent to this user base64-encoded in the `Pay` header, for example:
+
+```js
+amount = '1'
+shopAddress = 'test.crypto.xrp.rrhnXcox5bEmZfJCHzPxajUtwdt772zrCW'
+userId = 'UUxOFrawp0A'
+conditionSeed = Buffer.from('Agli74z_HjSNuTE1rTIr7QCkzWdIA3QdKoPMYHhw1I4', 'base64')
+xPayHeader = [ amount, shopAddress + '.' + userId, base64(conditionSeed) ]
+paymentId = '1'
+paymentPacketContents = {
+  amount: xPayHeader[0],
+  account: xPayHeader[1] + '.' + paymentId,
+  data: ''
+}
+fulfillment = hmac(conditionSeed, paymentPacketContents.account)
+condition = sha256(fulfillment)
+```
+
+To run this new version of the shop, type this into your terminal:
+
+```sh
+npm install ilp-packet
+node ./shop2.js
+```
+
+The ILP paacket is not very useful in a one-transfer payment, but in future tutorials, we see how connectors can forward an
+incoming transfer, and thus connect one ledger to other ledgers. When that happens, we say the whole sender
+to receiver process is the "payment", and each link in the chain is a "transfer", so one payment consists of 
+one or more transfers. The sender can then put information in the ILP packet which the receiver can use
+to generate the fulfillment, and if that information would be tempered with by connectors along the way,
+this would be detected because the fulfillment would not match.
+The ILP Payment Packet is serialized into OER and then Base64-encoded before it's added to the transfer as a Memo.
+Note that this assumes that the ledger supports adding a memo to the transfer, but luckily, most ledgers do.
```

Comment:
I'd think of it slightly differently. Most ledgers don't have the capability to attach a memo but since we require that functionality it can also be sent out of band (e.g. over the internet)